modules/apps/item-selector/item-selector-taglib/src/main/resources/META-INF/resources/item_selector_preview/js/ItemSelectorPreview.es.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {ClayIconSpriteContext} from '@clayui/icon';
+import dom from 'metal-dom';
+import PropTypes from 'prop-types';
+import React, {Component} from 'react';
+
+import Header from './Header.es';
+import Carousel from './Carousel.es';
+import Footer from './Footer.es';
+
+class ItemSelectorPreview extends Component {
+	static propTypes = {
+		handleSelectedItem: PropTypes.func.isRequired,
+		headerTitle: PropTypes.string.isRequired,
+		links: PropTypes.string.isRequired,
+		selector: PropTypes.string
+	};
+
+	constructor(props) {
+		super(props);
+
+		let items = Array.from(document.querySelectorAll(this.props.links));
+
+		this.state = {
+			currentItem: null,
+			currentItemIndex: 0,
+			items: items,
+			openViewer: false
+		};
+
+		const selector = this.props.selector;
+
+		items.forEach((item, index) => {
+			let cliclableItem = item;
+
+			if (selector) {
+				cliclableItem = item.querySelector(selector);
+			}
```

Comment:
Operations with the DOM should be performed inside lifecycle methods like `componentDidMount` or `componentDidUpdate`. Even though it works now, there is no guarantee that the DOM is ready when running the constructor. 

Also, there is a typo in `cliclable`.

Reply:
Good appreciation!